crates/solidity/outputs/cargo/crate/src/backend/l1/generated/mutator.rs

```diff
@@ -0,0 +1,2961 @@
+    ) -> ImportDeconstructionSymbol {
+        let name = Rc::clone(&source.name);
+        let alias = source
+            .alias
+            .as_ref()
+            .map(|value| self.mutate_import_alias(value));
+
+        Rc::new(ImportDeconstructionSymbolStruct {
+            node_id: source.node_id,
+            name,
+            alias,
+        })
+    }
+
+    fn mutate_import_alias(&mut self, source: &ImportAlias) -> ImportAlias {
+        let identifier = Rc::clone(&source.identifier);
+
+        Rc::new(ImportAliasStruct {
+            node_id: source.node_id,
+            identifier,
+        })
+    }
+
+    fn mutate_using_directive(&mut self, source: &UsingDirective) -> UsingDirective {
+        let clause = self.mutate_using_clause(&source.clause);
+        let target = self.mutate_using_target(&source.target);
+        let global_keyword = source.global_keyword.as_ref().map(Rc::clone);
+
+        Rc::new(UsingDirectiveStruct {
+            node_id: source.node_id,
+            clause,
+            target,
+            global_keyword,
+        })
+    }
+
+    fn mutate_using_deconstruction(&mut self, source: &UsingDeconstruction) -> UsingDeconstruction {
+        let symbols = self.mutate_using_deconstruction_symbols(&source.symbols);
+
+        Rc::new(UsingDeconstructionStruct {
+            node_id: source.node_id,
+            symbols,
+        })
+    }
+
+    fn mutate_using_deconstruction_symbol(
+        &mut self,
+        source: &UsingDeconstructionSymbol,
+    ) -> UsingDeconstructionSymbol {
+        let name = self.mutate_identifier_path(&source.name);
+        let alias = source
+            .alias
+            .as_ref()
+            .map(|value| self.mutate_using_alias(value));
+
+        Rc::new(UsingDeconstructionSymbolStruct {
+            node_id: source.node_id,
+            name,
+            alias,
+        })
+    }
+
+    fn mutate_using_alias(&mut self, source: &UsingAlias) -> UsingAlias {
+        let operator = self.mutate_using_operator(&source.operator);
+
+        Rc::new(UsingAliasStruct {
+            node_id: source.node_id,
+            operator,
+        })
+    }
+
+    fn mutate_contract_definition(&mut self, source: &ContractDefinition) -> ContractDefinition {
+        let abstract_keyword = source.abstract_keyword.as_ref().map(Rc::clone);
+        let name = Rc::clone(&source.name);
+        let members = self.mutate_contract_members(&source.members);
+        let inheritance_types = self.mutate_inheritance_types(&source.inheritance_types);
+        let storage_layout = source
+            .storage_layout
+            .as_ref()
+            .map(|value| self.mutate_storage_layout_specifier(value));
+
+        Rc::new(ContractDefinitionStruct {
+            node_id: source.node_id,
+            abstract_keyword,
+            name,
+            members,
+            inheritance_types,
+            storage_layout,
+        })
+    }
+
+    fn mutate_inheritance_specifier(
```

Comment:
Q: Since this is passed a readonly reference to the source node, and is expected to return a new owned value, WDYT of calling this feature "rewriter"?
I initially assumed that a "mutator" would receive a mutable node reference.